contracts/common/Registry.sol

```diff
@@ -16,10 +15,16 @@ contract Registry is Ownable {
   mapping(bytes32 => address) contractMap;
   mapping(address => address) public rootToChildToken;
   mapping(address => address) public childToRootToken;
-  mapping(address => bool) public proofValidatorContracts;
   // @todo we can think of one function from the registry which returns both (childToken,isERC721) if we are using it frequently together.
+  mapping(address => bool) public proofValidatorContracts;
   mapping(address => bool) public isERC721;
 
+  enum Type { Invalid, ERC20, ERC721 }
```

Comment:
It can be any type of predicate. If we want user to add any predicate, shouldn't we allow any type?

Reply:
+1, we can have something like a `custom` in enum but for now we should stay on 20/721 because of bindings of the predicates with `root chain` and `withdraw manager`.

Reply:
+1 for adding a `custom`. @0xAshish  As long as the predicate follows this interface for calling withdraw manager, we are fine.
```
function addExitToQueue(
    address exitor,
    address childToken,
    address rootToken,
    uint256 exitAmountOrTokenId,
    bytes32 txHash,
    bool burnt,
    uint256 priority)
```

Reply:
For custom predicates, we need to map addresses.

Reply:
Yes, see `addPredicate` in `registry.sol`. And `addExitToQueue ` is ACLed on `isPredicateAuthorized`